src/averageAges.js

```diff
@@ -55,10 +73,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people
+    .filter(({ sex, mother }) => people.some(({ name }) => name === mother)
+      && (onlyWithSon ? sex === MALE_SEX : true)
+    );
```

Comment:
Now everything is written correctly, but it is so difficult to read and expand. Break it into smaller parts.